src/note_ends.lua

```diff
@@ -60,6 +60,7 @@ function expand_note_ends()
             and entry:Next():IsRest() -- that is a rest
             and entry.Duration < note_value -- this note is too short
             and note_boundary == 0 -- on a beat or half-beat
+            and entry.Duration + entry:Next().Duration >= note_value
             and
             (    (beat_duration >= position_in_beat + note_value) -- enough space for expanded endpoint
```

Comment:
This is a style issue, but it is important. You should *never* end up with two `end` statements vertically aligned. It is a recipe for a long frustrating slog sometime in the future. I would tab the entire `then`...`end` block over one to the right.